README.md

```diff
@@ -111,30 +111,52 @@ T::Coerce[Params].new.from({id: '1'})
 ```
 More examples: [nested params](https://github.com/chanzuckerberg/sorbet-coerce/blob/a56c0c6a363bb49b11e77ac57893afc3d54c6b8c/spec/nested_spec.rb#L18-L26)
 
-## Coercion Error
+## Errors
+- `T::Coerce::CoercionError`: It fails to convert a value into the specified type (i.e. `'bad string args' to Integer`).
+- `T::Coerce::ShapeError`: The shape of the input does not match the shape of input type (i.e. `'1' to T::Array[Integer]`).
+- `TypeError`: The coerced input does not match input type.
 
-Sorbet-coerce throws a coercion error when it fails to convert a value into the specified type. The error is [configurable](https://sorbet.org/docs/runtime#changing-the-runtime-behavior) through `T::Configuration`. In an environment where type errors are configured to be silent (referred to soft errors), when the coercion fails (or constructing `T::Struct` fails), `T::Coerce` will return the original value instead of actually raising the errors (referred to hard errors).
+`TypeError`s are [configurable](https://sorbet.org/docs/runtime#changing-the-runtime-behavior) through `T::Configuration`. 
+`T::Coerce::CoercionError`s are configurable too:
+```ruby
+T::Coerce::Configuration.raise_coercion_error = false # default to true
+```
+We use an inline flag to overwrite the global configuration:
+```ruby
+T::Coerce[T.nilable(Integer)].new.from('abc', raise_coercion_error: false)
+```
+
+In an environment where type errors and coercion errors are configured to be silent (referred to soft errors), when the coercion fails, `T::Coerce` will fill the result with `nil` instead of actually raising the errors (referred to hard errors).
```

Comment:
```suggestion
In an environment where type errors and coercion errors are configured to be silent (referred to as soft errors), when the coercion fails, `T::Coerce` will fill the result with `nil` instead of actually raising the errors (referred to hard errors).
```

Reply:
I think this sentence should be moved to earlier in the doc. It's more important to explain the intention/the impact of the configuration than the detail of how to do it.

Reply:
When sorbet error is skipped, it doesn't fill `nil` to the value, it usually tries to use the given value right?

eg x = T.let(5, String) # in case of soft error we would get 5

Reply:
> eg x = T.let(5, String) # in case of soft error we would get 5
Yep.